metaflow/plugins/aws/aws_utils.py

```diff
@@ -4,6 +4,22 @@
 from metaflow.exception import MetaflowException
 
 
+def parse_s3_full_path(s3_uri):
+    from urllib.parse import urlparse
+
+    #  <scheme>://<netloc>/<path>;<params>?<query>#<fragment>
+    scheme, netloc, path, _, _, _ = urlparse(s3_uri)
+    assert scheme == "s3"
```

Comment:
within metaflow code base, we have elected to not use assert statements. also, when the assert statement fails, the user is left no better in their understanding of how to fix the issue

Reply:
This is copied from `parse_gcs_full_path`.. Happy to remove it though if we don't want it!